pkgs/development/tools/kustomize/default.nix

```
@@ -1,29 +1,29 @@
-{ lib, buildGoModule, fetchFromGitHub }:
+{ lib, buildGoModule, fetchFromGitHub, tree }:
 
 buildGoModule rec {
   pname = "kustomize";
-  version = "3.1.0";
-  # rev is the 3.1.0 commit, mainly for kustomize version command output
-  rev = "95f3303493fdea243ae83b767978092396169baf";
+  version = "3.3.1";
+  # rev is the 3.3.1 commit, mainly for kustomize version command output
+  # rev = "f2ac5a2d0df13c047fb20cbc12ef1a3b41ce2dad";
 
-  goPackagePath = "sigs.k8s.io/kustomize";
-  subPackages = [ "cmd/kustomize" ];
-
-  buildFlagsArray = let t = "${goPackagePath}/v3/pkg/commands/misc"; in ''
-    -ldflags=
-      -s -X ${t}.kustomizeVersion=${version}
-         -X ${t}.gitCommit=${rev}
-         -X ${t}.buildDate=unknown
-  '';
+  # they changed how the version is injected, not sure how to bump the version
+  #buildFlagsArray = let t = "sigs.k8s.io/kustomize/v3/pkg/commands/misc"; in ''
+  #  -ldflags=
+  #    -s -X sigs.k8s.io/kustomize/kustomize/v3/internal/commands.version=${version}
+  #       -X ${t}.gitCommit=${rev}
+  #       -X ${t}.buildDate=unknown
+  #'';
 
   src = fetchFromGitHub {
-    sha256 = "0kigcirkjvnj3xi1p28p9yp3s0lff24q5qcvf8ahjwvpbwka14sh";
+    sha256 = "0yxxz0b56r18w178y32s619zy8ci6l93c6vlzx11hhxhbw43f6v6";
     rev = "v${version}";
     repo = pname;
     owner = "kubernetes-sigs";
   };
 
-  modSha256 = "0w8sp73pmj2wqrg7x7z8diglyfq6c6gn9mmck0k1gk90nv7s8rf1";
+  sourceRoot = "source/kustomize";
+
+  modSha256 = "0yypvhc4ain2zf1zc0g2jk2rng3c7js2nn31rf358s46r2q5f31g";
```

This doesn't match for me locally. Instead this seems to be correct checksum.

```suggestion
  modSha256 = "0f8h8ipg7miaz5waniybl2wnp3pfv59yrd065nz418ckj1a21qw2";
```

---

this makes sense, I think i did this before i changed sourceRoot

---

I'm honesly not really sure. I have not enough go on nix experience. I'm not the biggest fan of golang myself :speak_no_evil: I'm just trying to repay you for reviews on my PRs <3 in a role of rubber :duck:

---

I appreciate it! :)